### ISSUE TITLE
Remove `py` dependency?

### ISSUE THREAD

**Issue by janbrasna** [mozilla/bedrock#14851]
## Description

There's still `py==1.11.0` required in dev.in, which is basically abandonware these days. Not updated for years, vulnerability reported even for the latest version.

Is it really being used anywhere?
_(Or am I overlooking something too obvious?)_

(The legacy `py.test` entrypoints of `pytest` should not be influenced by this in any way IIRC.)

---

#### Success Criteria
- [ ] Auditing the use of `py` package
- [ ] Removing it if not needed anymore

**Comment by janbrasna** [comment]
FYI `py` is not needed for `py.test` (cli) to work, it has been independent on it completely for years (several major versions):

- https://github.com/pytest-dev/pytest/issues/1629
- https://github.com/pytest-dev/pytest/issues/1632
- https://github.com/pytest-dev/pytest/pull/1633
- [entrypoints](https://github.com/pytest-dev/pytest/blob/79de84941a50eda7a6f0511cb0ce7a63776e4563/setup.cfg#L64) definition
- [legacy test](https://github.com/pytest-dev/pytest/blob/3433ba4fa4d2b971f762628f98c88cc426417226/testing/test_entry_points.py#L10) (to ensure both are equivalent)

But good point @stevejalim to also update the entrypoints. They are nonetheless not blocking the removal of the dep.

**Comment by janbrasna** [comment]
This is blocked on not only removing (or upgrading) `pytest-selenium`, but also migrating away from `pytest-parallel` — see #14867 for more details.
